scripts/runPathSimulator.ts

```diff
@@ -0,0 +1,254 @@
+// tslint:disable:no-console
+import express = require('express')
+import WebSocket = require('ws')
+import http = require('http')
+
+const url = require('url')
+
+// ******************************************************************************************************* //
+// ***** Movements ***** //
+// ******************************************************************************************************* //
+
+const Sense = {
+  POSITIVE: step => (current, target) => {
+    const next = current + step
+    return next >= target ? target : next
+  },
+  NEGATIVE: step => (current, target) => {
+    const next = current - step
+    return next <= target ? target : next
+  }
+}
+
+function single(...moves) {
+  return function*() {
+    for (const move of moves) {
+      yield* move()
+    }
+  }
+}
+
+function loop(...moves) {
+  return function*() {
+    while (true) {
+      for (const move of moves) {
+        yield* move()
+      }
+    }
+  }
+}
+
+function logOut() {
+  return function*() {
+    return { x: 16 * 1000, y: 1000, z: 16 * 1000 }
+  }
+}
+
+function walkTo(x: number, z: number) {
+  return function*() {
+    let position = yield
+
+    const xSense = (x > position.x ? Sense.POSITIVE : Sense.NEGATIVE)(2)
```

Comment:
do you think that this might be more legible?
```suggestion
    const moveTowardsTargetX = (x > position.x ? Sense.POSITIVE : Sense.NEGATIVE)(2)
```